models/order.js

```diff
@@ -32,6 +41,27 @@ function getOrders() {
   });
 }
 
+function updateEntry(orderId, entryIndex, update) {
+  const Order = this;
+
+  return new Promise((resolve, reject) => {
+    Order.findOne({ id: orderId })
+      .then((order) => {
+        const entries = order.entries;
+        const entry = entries[entryIndex];
+        const updatedEntries = [
+          ...entries.slice(0, entryIndex),
+          _.extend(entry, update),
```

Comment:
try _.extend({}, entry, update). The way you are doing it now i think would just update entry.